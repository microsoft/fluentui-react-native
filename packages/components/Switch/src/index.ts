export type { SwitchTokens, SwitchProps, SwitchState, SwitchSlotProps, SwitchType } from './Switch.types';
export { Switch } from './Switch';
export { useSwitch } from './useSwitch';
