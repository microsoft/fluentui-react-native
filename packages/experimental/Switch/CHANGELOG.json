{
  "name": "@fluentui-react-native/switch",
  "entries": [
    {
      "date": "Fri, 26 Aug 2022 18:15:17 GMT",
      "tag": "@fluentui-react-native/switch_v0.6.2",
      "version": "0.6.2",
      "comments": {
        "patch": [
          {
            "author": "nkhalil942@gmail.com",
            "package": "@fluentui-react-native/switch",
            "commit": "aa020f745c509bfda861b349b0ab89e7e78ebc8e",
            "comment": "Added better HC token mapping for win32"
          }
        ]
      }
    },
    {
      "date": "Fri, 19 Aug 2022 15:32:12 GMT",
      "tag": "@fluentui-react-native/switch_v0.6.1",
      "version": "0.6.1",
      "comments": {
        "patch": [
          {
            "author": "nkhalil942@gmail.com",
            "package": "@fluentui-react-native/switch",
            "commit": "c91a22dde35ebcc2f3de8fe4a1dc2e7959692594",
            "comment": "Added more space between the On/Off text, Label, and Track to align more with Fluent v9."
          }
        ]
      }
    },
    {
      "date": "Wed, 17 Aug 2022 18:45:52 GMT",
      "tag": "@fluentui-react-native/switch_v0.6.0",
      "version": "0.6.0",
      "comments": {
        "minor": [
          {
            "author": "nkhalil942@gmail.com",
            "package": "@fluentui-react-native/switch",
            "commit": "0c9ec57d22d8524478abdf49b4fea2d8e93bcb56",
            "comment": " Label position set to 'after' will prevent onText and offText from rendering"
          }
        ]
      }
    },
    {
      "date": "Thu, 11 Aug 2022 22:49:12 GMT",
      "tag": "@fluentui-react-native/switch_v0.5.0",
      "version": "0.5.0",
      "comments": {
        "minor": [
          {
            "author": "duomaster01@gmail.com",
            "package": "@fluentui-react-native/switch",
            "commit": "25f9f2b38dc274bb43b5231c4a6ee4d88a555ce9",
            "comment": "Replaced LayoutAnimation with animationClass"
          }
        ]
      }
    },
    {
      "date": "Wed, 10 Aug 2022 18:25:08 GMT",
      "tag": "@fluentui-react-native/switch_v0.4.1",
      "version": "0.4.1",
      "comments": {
        "patch": [
          {
            "author": "nkhalil942@gmail.com",
            "package": "@fluentui-react-native/switch",
            "commit": "2aa1ff6d9d87c5e5eec9fe184c37f24ee1c9951b",
            "comment": "Added Switch Documentation"
          }
        ]
      }
    },
    {
      "date": "Mon, 08 Aug 2022 21:40:41 GMT",
      "tag": "@fluentui-react-native/switch_v0.4.0",
      "version": "0.4.0",
      "comments": {
        "minor": [
          {
            "author": "nkhalil942@gmail.com",
            "package": "@fluentui-react-native/switch",
            "commit": "abeef208893a37feffab338d8074de3e064c7703",
            "comment": "Added token customization"
          }
        ]
      }
    },
    {
      "date": "Fri, 05 Aug 2022 23:22:38 GMT",
      "tag": "@fluentui-react-native/switch_v0.3.1",
      "version": "0.3.1",
      "comments": {
        "patch": [
          {
            "author": "beachball",
            "package": "@fluentui-react-native/switch",
            "comment": "Bump @fluentui-react-native/text to v0.14.0",
            "commit": "b77d041fc8feb1088d2b4f1a1a755a9061148b97"
          },
          {
            "author": "beachball",
            "package": "@fluentui-react-native/switch",
            "comment": "Bump @fluentui-react-native/interactive-hooks to v0.18.1",
            "commit": "b77d041fc8feb1088d2b4f1a1a755a9061148b97"
          }
        ]
      }
    },
    {
      "date": "Thu, 04 Aug 2022 19:24:58 GMT",
      "tag": "@fluentui-react-native/switch_v0.3.0",
      "version": "0.3.0",
      "comments": {
        "minor": [
          {
            "author": "nkhalil942@gmail.com",
            "package": "@fluentui-react-native/switch",
            "commit": "0628c88875ece4df49bb7893dd35a524eccf900a",
            "comment": "Added Switch control for Win32"
          }
        ]
      }
    },
    {
      "date": "Wed, 03 Aug 2022 19:25:34 GMT",
      "tag": "@fluentui-react-native/switch_v0.2.0",
      "version": "0.2.0",
      "comments": {
        "minor": [
          {
            "author": "nkhalil942@gmail.com",
            "package": "@fluentui-react-native/switch",
            "commit": "0628c88875ece4df49bb7893dd35a524eccf900a",
            "comment": "Added Switch control for Win32"
          }
        ]
      }
    }
  ]
}
