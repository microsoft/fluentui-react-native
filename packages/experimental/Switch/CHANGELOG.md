# Change Log - @fluentui-react-native/switch

This log was last generated on Wed, 25 Jan 2023 21:31:18 GMT and should not be manually modified.

<!-- Start content -->

## 0.8.19

Wed, 25 Jan 2023 21:31:18 GMT

### Patches

- switch initial animation fix (rohanpd.work@gmail.com)
- Bump @fluentui-react-native/text to v0.19.15
- Bump @fluentui-react-native/framework to v0.8.36
- Bump @fluentui-react-native/interactive-hooks to v0.22.13
- Bump @fluentui-react-native/tokens to v0.20.5

## 0.8.18

Fri, 20 Jan 2023 21:49:21 GMT

### Patches

- Bump @fluentui-react-native/text to v0.19.14

## 0.8.17

Fri, 20 Jan 2023 18:06:46 GMT

### Patches

- Bump @fluentui-react-native/text to v0.19.13
- Bump @fluentui-react-native/framework to v0.8.35
- Bump @fluentui-react-native/interactive-hooks to v0.22.12

## 0.8.16

Wed, 18 Jan 2023 01:55:11 GMT

### Patches

- Bump @fluentui-react-native/text to v0.19.12
- Bump @fluentui-react-native/framework to v0.8.34
- Bump @fluentui-react-native/interactive-hooks to v0.22.11
- Bump @fluentui-react-native/tokens to v0.20.4

## 0.8.15

Tue, 10 Jan 2023 20:17:18 GMT

### Patches

- Bump @fluentui-react-native/text to v0.19.11
- Bump @fluentui-react-native/framework to v0.8.33
- Bump @fluentui-react-native/interactive-hooks to v0.22.10
- Bump @fluentui-react-native/tokens to v0.20.3

## 0.8.14

Tue, 10 Jan 2023 06:10:57 GMT

### Patches

- add switch examples (rohanpd.work@gmail.com)

## 0.8.13

Tue, 10 Jan 2023 05:10:08 GMT

### Patches

- switch disable state a11y fix (rohanpd.work@gmail.com)

## 0.8.12

Mon, 09 Jan 2023 22:25:00 GMT

### Patches

- Bump @fluentui-react-native/text to v0.19.10
- Bump @fluentui-react-native/framework to v0.8.32
- Bump @fluentui-react-native/interactive-hooks to v0.22.9

## 0.8.11

Fri, 06 Jan 2023 12:26:31 GMT

### Patches

- revert customised checked file (rohanpd.work@gmail.com)
- Bump @fluentui-react-native/text to v0.19.9
- Bump @fluentui-react-native/framework to v0.8.31
- Bump @fluentui-react-native/interactive-hooks to v0.22.8

## 0.8.10

Wed, 04 Jan 2023 16:55:53 GMT

### Patches

- Bump @fluentui-react-native/text to v0.19.8
- Bump @fluentui-react-native/interactive-hooks to v0.22.7

## 0.8.9

Tue, 03 Jan 2023 23:38:40 GMT

### Patches

- Bump @fluentui-react-native/text to v0.19.7

## 0.8.8

Fri, 30 Dec 2022 16:09:27 GMT

### Patches

- Bump @fluentui-react-native/text to v0.19.6
- Bump @fluentui-react-native/framework to v0.8.30
- Bump @fluentui-react-native/interactive-hooks to v0.22.6

## 0.8.7

Fri, 30 Dec 2022 03:49:05 GMT

### Patches

- Bump @fluentui-react-native/text to v0.19.5
- Bump @fluentui-react-native/interactive-hooks to v0.22.5

## 0.8.6

Thu, 29 Dec 2022 08:02:57 GMT

### Patches

- Bump @fluentui-react-native/text to v0.19.4

## 0.8.5

Thu, 29 Dec 2022 03:47:47 GMT

### Patches

- Bump @fluentui-react-native/text to v0.19.3
- Bump @fluentui-react-native/framework to v0.8.29
- Bump @fluentui-react-native/interactive-hooks to v0.22.4
- Bump @fluentui-react-native/tokens to v0.20.2

## 0.8.4

Tue, 27 Dec 2022 22:21:15 GMT

### Patches

- Bump @fluentui-react-native/text to v0.19.2
- Bump @fluentui-react-native/framework to v0.8.28
- Bump @fluentui-react-native/interactive-hooks to v0.22.3
- Bump @fluentui-react-native/tokens to v0.20.1

## 0.8.3

Tue, 27 Dec 2022 10:18:14 GMT

### Patches

- Update token values for android. (email not defined)
- Bump @fluentui-react-native/text to v0.19.1
- Bump @fluentui-react-native/framework to v0.8.27
- Bump @fluentui-react-native/interactive-hooks to v0.22.2

## 0.8.2

Fri, 23 Dec 2022 05:52:48 GMT

### Patches

- doc update for switch - android (rohanpd.work@gmail.com)

## 0.8.1

Thu, 15 Dec 2022 14:50:50 GMT

### Patches

- Bump @fluentui-react-native/text to v0.19.0
- Bump @fluentui-react-native/framework to v0.8.26
- Bump @fluentui-react-native/interactive-hooks to v0.22.1
- Bump @fluentui-react-native/tokens to v0.20.0

## 0.8.0

Fri, 09 Dec 2022 05:23:38 GMT

### Minor changes

- switch initial tokenization and tester app changes (rohanpd.work@gmail.com)
- Bump @fluentui-react-native/text to v0.18.2
- Bump @fluentui-react-native/framework to v0.8.25
- Bump @fluentui-react-native/interactive-hooks to v0.22.0
- Bump @fluentui-react-native/tokens to v0.19.1

## 0.7.9

Tue, 06 Dec 2022 16:59:54 GMT

### Patches

- Bump @fluentui-react-native/text to v0.18.1
- Bump @fluentui-react-native/framework to v0.8.24
- Bump @fluentui-react-native/interactive-hooks to v0.21.5

## 0.7.8

Tue, 06 Dec 2022 00:46:41 GMT

### Patches

- Bump @fluentui-react-native/text to v0.18.0
- Bump @fluentui-react-native/framework to v0.8.23
- Bump @fluentui-react-native/interactive-hooks to v0.21.4
- Bump @fluentui-react-native/tokens to v0.19.0

## 0.7.7

Thu, 01 Dec 2022 03:12:20 GMT

### Patches

- Bump @fluentui-react-native/text to v0.17.1
- Bump @fluentui-react-native/framework to v0.8.22
- Bump @fluentui-react-native/interactive-hooks to v0.21.3
- Bump @fluentui-react-native/tokens to v0.18.1

## 0.7.6

Mon, 28 Nov 2022 19:49:35 GMT

### Patches

- Bump @fluentui-react-native/text to v0.17.0

## 0.7.5

Mon, 28 Nov 2022 11:51:06 GMT

### Patches

- Bump @fluentui-react-native/text to v0.16.2
- Bump @fluentui-react-native/framework to v0.8.21
- Bump @fluentui-react-native/interactive-hooks to v0.21.2
- Bump @fluentui-react-native/tokens to v0.18.0

## 0.7.4

Wed, 16 Nov 2022 08:47:25 GMT

### Patches

- Bump @fluentui-react-native/text to v0.16.1
- Bump @fluentui-react-native/framework to v0.8.20
- Bump @fluentui-react-native/interactive-hooks to v0.21.1

## 0.7.3

Sun, 13 Nov 2022 09:17:07 GMT

### Patches

- Bump @fluentui-react-native/text to v0.16.0
- Bump @fluentui-react-native/interactive-hooks to v0.21.0

## 0.7.2

Fri, 11 Nov 2022 18:06:32 GMT

### Patches

- Bump @fluentui-react-native/text to v0.15.15
- Bump @fluentui-react-native/framework to v0.8.19
- Bump @fluentui-react-native/interactive-hooks to v0.20.2

## 0.7.1

Wed, 09 Nov 2022 22:52:11 GMT

### Patches

- Bump @fluentui-react-native/interactive-hooks to v0.20.1

## 0.7.0

Tue, 08 Nov 2022 22:18:44 GMT

### Minor changes

- update @office-iss/react-native-win32 to 0.68.8 (krsiler@microsoft.com)
- Bump @fluentui-react-native/adapters to v0.10.0
- Bump @fluentui-react-native/text to v0.15.14
- Bump @fluentui-react-native/framework to v0.8.18
- Bump @fluentui-react-native/interactive-hooks to v0.20.0
- Bump @fluentui-react-native/tokens to v0.17.8

## 0.6.27

Tue, 08 Nov 2022 19:22:01 GMT

### Patches

- Bump @fluentui-react-native/text to v0.15.13
- Bump @fluentui-react-native/framework to v0.8.17
- Bump @fluentui-react-native/interactive-hooks to v0.19.7

## 0.6.26

Fri, 04 Nov 2022 14:36:23 GMT

### Patches

- Bump @fluentui-react-native/text to v0.15.12
- Bump @fluentui-react-native/framework to v0.8.16
- Bump @fluentui-react-native/interactive-hooks to v0.19.6
- Bump @fluentui-react-native/tokens to v0.17.7

## 0.6.25

Tue, 01 Nov 2022 16:30:56 GMT

### Patches

- Bump @fluentui-react-native/interactive-hooks to v0.19.5

## 0.6.24

Sat, 29 Oct 2022 01:05:20 GMT

### Patches

- MenuV1: Update types to PressablePropsExtended (sanajmi@microsoft.com)

## 0.6.23

Thu, 27 Oct 2022 11:09:35 GMT

### Patches

- Bump @fluentui-react-native/text to v0.15.11
- Bump @fluentui-react-native/framework to v0.8.15
- Bump @fluentui-react-native/interactive-hooks to v0.19.4
- Bump @fluentui-react-native/tokens to v0.17.6

## 0.6.22

Tue, 25 Oct 2022 22:47:45 GMT

### Patches

- Bump @fluentui-react-native/text to v0.15.10
- Bump @fluentui-react-native/framework to v0.8.14
- Bump @fluentui-react-native/interactive-hooks to v0.19.3

## 0.6.21

Fri, 21 Oct 2022 13:09:26 GMT

### Patches

- Bump @fluentui-react-native/text to v0.15.9
- Bump @fluentui-react-native/framework to v0.8.13
- Bump @fluentui-react-native/interactive-hooks to v0.19.2

## 0.6.20

Thu, 20 Oct 2022 18:30:02 GMT

### Patches

- Bump @fluentui-react-native/interactive-hooks to v0.19.1

## 0.6.19

Tue, 18 Oct 2022 04:06:27 GMT

### Patches

- Bump @fluentui-react-native/interactive-hooks to v0.19.0

## 0.6.18

Mon, 17 Oct 2022 23:24:26 GMT

### Patches

- Switch Switch to Pressable (sanajmi@microsoft.com)

## 0.6.17

Mon, 17 Oct 2022 17:52:49 GMT

### Patches

- Bump @fluentui-react-native/text to v0.15.8
- Bump @fluentui-react-native/framework to v0.8.12
- Bump @fluentui-react-native/interactive-hooks to v0.18.15

## 0.6.16

Fri, 14 Oct 2022 19:11:13 GMT

### Patches

- Bump @fluentui-react-native/adapters to v0.9.3
- Bump @fluentui-react-native/text to v0.15.7
- Bump @fluentui-react-native/framework to v0.8.11
- Bump @fluentui-react-native/interactive-hooks to v0.18.14
- Bump @fluentui-react-native/tokens to v0.17.5

## 0.6.15

Wed, 12 Oct 2022 21:54:15 GMT

### Patches

- Bump @fluentui-react-native/text to v0.15.6
- Bump @fluentui-react-native/framework to v0.8.10
- Bump @fluentui-react-native/interactive-hooks to v0.18.13
- Bump @fluentui-react-native/tokens to v0.17.4

## 0.6.14

Tue, 11 Oct 2022 22:41:44 GMT

### Patches

- Bump @fluentui-react-native/text to v0.15.5
- Bump @fluentui-react-native/framework to v0.8.9
- Bump @fluentui-react-native/interactive-hooks to v0.18.12
- Bump @fluentui-react-native/tokens to v0.17.3

## 0.6.13

Thu, 06 Oct 2022 17:24:49 GMT

### Patches

- Bump @fluentui-react-native/text to v0.15.4
- Bump @fluentui-react-native/framework to v0.8.8
- Bump @fluentui-react-native/interactive-hooks to v0.18.11

## 0.6.12

Tue, 04 Oct 2022 21:09:28 GMT

### Patches

- Bump @fluentui-react-native/text to v0.15.3
- Bump @fluentui-react-native/framework to v0.8.7
- Bump @fluentui-react-native/interactive-hooks to v0.18.10

## 0.6.11

Fri, 30 Sep 2022 08:04:44 GMT

### Patches

- Bump @fluentui-react-native/text to v0.15.2
- Bump @fluentui-react-native/framework to v0.8.6
- Bump @fluentui-react-native/interactive-hooks to v0.18.9
- Bump @fluentui-react-native/tokens to v0.17.2

## 0.6.10

Fri, 30 Sep 2022 00:54:36 GMT

### Patches

- Update react-native to 0.68 (krsiler@microsoft.com)
- Bump @fluentui-react-native/adapters to v0.9.2
- Bump @fluentui-react-native/text to v0.15.1
- Bump @fluentui-react-native/framework to v0.8.5
- Bump @fluentui-react-native/interactive-hooks to v0.18.8
- Bump @fluentui-react-native/tokens to v0.17.1
- Bump @fluentui-react-native/use-styling to v0.9.1

## 0.6.9

Mon, 26 Sep 2022 02:39:40 GMT

### Patches

- Bump @fluentui-react-native/interactive-hooks to v0.18.7

## 0.6.8

Wed, 21 Sep 2022 21:00:40 GMT

### Patches

- Bump @fluentui-react-native/text to v0.15.0
- Bump @fluentui-react-native/framework to v0.8.4
- Bump @fluentui-react-native/interactive-hooks to v0.18.6
- Bump @fluentui-react-native/tokens to v0.17.0

## 0.6.7

Tue, 20 Sep 2022 16:58:04 GMT

### Patches

- Updated SwitchProp comments to better align with the prop comments on the Fluent UI React Switch documentation. Did not make any functional changes to the control, but wanted to update documentation to get it more ready for publishing as a V1 control. (ejlayne@microsoft.com)

## 0.6.6

Fri, 16 Sep 2022 01:53:06 GMT

### Patches

- Bump @fluentui-react-native/text to v0.14.4
- Bump @fluentui-react-native/framework to v0.8.3
- Bump @fluentui-react-native/interactive-hooks to v0.18.5
- Bump @fluentui-react-native/tokens to v0.16.2

## 0.6.5

Wed, 14 Sep 2022 23:07:41 GMT

### Patches

- Remove rn win32 dep (ruaraki@microsoft.com)
- Bump @fluentui-react-native/adapters to v0.9.1
- Bump @fluentui-react-native/text to v0.14.3
- Bump @fluentui-react-native/framework to v0.8.2
- Bump @fluentui-react-native/interactive-hooks to v0.18.4
- Bump @fluentui-react-native/tokens to v0.16.1

## 0.6.4

Tue, 06 Sep 2022 18:22:03 GMT

### Patches

- Bump @fluentui-react-native/text to v0.14.2
- Bump @fluentui-react-native/interactive-hooks to v0.18.3

## 0.6.3

Mon, 29 Aug 2022 23:21:44 GMT

### Patches

- Bump @fluentui-react-native/text to v0.14.1
- Bump @fluentui-react-native/framework to v0.8.1
- Bump @fluentui-react-native/interactive-hooks to v0.18.2

## 0.6.2

Fri, 26 Aug 2022 18:15:17 GMT

### Patches

- Added better HC token mapping for win32 (nkhalil942@gmail.com)

## 0.6.1

Fri, 19 Aug 2022 15:32:12 GMT

### Patches

- Added more space between the On/Off text, Label, and Track to align more with Fluent v9. (nkhalil942@gmail.com)

## 0.6.0

Wed, 17 Aug 2022 18:45:52 GMT

### Minor changes

-  Label position set to 'after' will prevent onText and offText from rendering (nkhalil942@gmail.com)

## 0.5.0

Thu, 11 Aug 2022 22:49:12 GMT

### Minor changes

- Replaced LayoutAnimation with animationClass (duomaster01@gmail.com)

## 0.4.1

Wed, 10 Aug 2022 18:25:08 GMT

### Patches

- Added Switch Documentation (nkhalil942@gmail.com)

## 0.4.0

Mon, 08 Aug 2022 21:40:41 GMT

### Minor changes

- Added token customization (nkhalil942@gmail.com)

## 0.3.1

Fri, 05 Aug 2022 23:22:38 GMT

### Patches

- Bump @fluentui-react-native/text to v0.14.0
- Bump @fluentui-react-native/interactive-hooks to v0.18.1

## 0.3.0

Thu, 04 Aug 2022 19:24:58 GMT

### Minor changes

- Added Switch control for Win32 (nkhalil942@gmail.com)

## 0.2.0

Wed, 03 Aug 2022 19:25:34 GMT

### Minor changes

- Added Switch control for Win32 (nkhalil942@gmail.com)
