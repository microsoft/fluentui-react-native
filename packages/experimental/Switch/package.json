{
  "name": "@fluentui-react-native/switch",
  "version": "0.6.2",
  "description": "A cross-platform Switch component using the Fluent Design System",
  "main": "src/index.ts",
  "module": "src/index.ts",
  "typings": "lib/index.d.ts",
  "onPublish": {
    "main": "lib-commonjs/index.js",
    "module": "lib/index.js"
  },
  "scripts": {
    "build": "fluentui-scripts build",
    "depcheck": "fluentui-scripts depcheck",
    "just": "fluentui-scripts",
    "clean": "fluentui-scripts clean",
    "lint": "fluentui-scripts eslint",
    "test": "fluentui-scripts jest",
    "update-snapshots": "fluentui-scripts jest -u",
    "prettier": "fluentui-scripts prettier",
    "prettier-fix": "fluentui-scripts prettier --fix true"
  },
  "repository": {
    "type": "git",
    "url": "https://github.com/microsoft/fluentui-react-native.git",
    "directory": "packages/experimental/Switch"
  },
  "dependencies": {
    "@fluentui-react-native/adapters": ">=0.9.0 <1.0.0",
    "@fluentui-react-native/text": ">=0.14.0 <1.0.0",
    "@fluentui-react-native/framework": "0.8.0",
    "@fluentui-react-native/interactive-hooks": ">=0.18.1 <1.0.0",
    "@fluentui-react-native/tokens": ">=0.16.0 <1.0.0",
    "@fluentui-react-native/use-styling": ">=0.9.0 <1.0.0",
    "tslib": "^2.3.1"
  },
  "devDependencies": {
    "@fluentui-react-native/eslint-config-rules": "^0.1.1",
    "@fluentui-react-native/test-tools": ">=0.1.1 <1.0.0",
    "@office-iss/react-native-win32": "^0.66.0",
    "@types/react-native": "^0.66.0",
    "@fluentui-react-native/scripts": "^0.1.1",
    "react": "17.0.2",
    "react-native": "^0.66.0"
  },
  "peerDependencies": {
    "react": ">=17.0.2",
    "react-native": ">=0.66.0"
  },
  "author": "",
  "license": "MIT"
}
