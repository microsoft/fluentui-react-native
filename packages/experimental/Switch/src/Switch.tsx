/** @jsx withSlots */
import { Pressable, View } from 'react-native';
import { Text } from '@fluentui-react-native/text';
import { switchName, SwitchType, SwitchProps } from './Switch.types';
import { compose } from '@fluentui-react-native/framework';

export const Switch = compose<SwitchType>({
  displayName: switchName,
  slots: {
    root: Pressable,
    label: Text,
    track: View,
    thumb: View,
    toggleContainer: View,
    onOffText: Text,
  },
  useRender: (_userProps: SwitchProps) => {
    return (_final: SwitchProps) => {
      return null; // Only implemented for Win32
    };
  },
});
