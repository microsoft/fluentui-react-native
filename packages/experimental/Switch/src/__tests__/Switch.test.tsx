import * as React from 'react';

import { checkReRender } from '@fluentui-react-native/test-tools';
import * as renderer from 'react-test-renderer';

import { Switch } from '../Switch';

// mocks out setTimeout and other timer functions with mock functions , test will fail without this as we're using Animated API
jest.useFakeTimers();

it('Switch Default', () => {
  const tree = renderer.create(<Switch label="Default Switch" />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Switch Disabled', () => {
  const tree = renderer.create(<Switch label="Default Switch Disabled" disabled />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Switch re-renders correctly', () => {
  checkReRender(() => <Switch label="Render Twice" />, 2);
});
