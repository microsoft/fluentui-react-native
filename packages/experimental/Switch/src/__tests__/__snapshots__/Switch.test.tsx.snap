// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch Default 1`] = `
<View
  accessibilityActions={
    Array [
      Object {
        "name": "Toggle",
      },
    ]
  }
  accessibilityLabel="Default Switch"
  accessibilityRole="switch"
  accessibilityState={
    Object {
      "checked": undefined,
      "disabled": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onAccessibilityAction={[Function]}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "borderColor": undefined,
      "borderRadius": undefined,
      "borderWidth": undefined,
      "flexDirection": "row-reverse",
      "minHeight": undefined,
      "minWidth": undefined,
    }
  }
  switchAnimationStyles={
    Object {
      "thumbAnimatedStyle": Object {
        "transform": Array [
          Object {
            "translateX": 0,
          },
        ],
      },
      "trackBackgroundStyle": Object {
        "backgroundColor": "rgba(15, 108, 189, 1)",
      },
    }
  }
>
  <Text
    style={
      Object {
        "color": "#242424",
        "fontFamily": "Segoe UI",
        "fontSize": 13,
        "fontWeight": "400",
        "margin": 0,
      }
    }
  >
    Default Switch
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row-reverse",
      }
    }
  >
    <View
      collapsable={false}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(15, 108, 189, 1)",
          "borderRadius": 100,
          "flexDirection": "row",
          "height": 32,
          "marginRight": 8,
          "width": 52,
        }
      }
    >
      <View
        animationClass="Ribbon_SwitchThumb"
        collapsable={false}
        style={
          Object {
            "backgroundColor": "#ffffff",
            "borderRadius": 100,
            "elevation": 8,
            "height": 26,
            "margin": 3,
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
            "width": 26,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`Switch Disabled 1`] = `
<View
  accessibilityActions={
    Array [
      Object {
        "name": "Toggle",
      },
    ]
  }
  accessibilityLabel="Default Switch Disabled"
  accessibilityRole="switch"
  accessibilityState={
    Object {
      "checked": undefined,
      "disabled": true,
    }
  }
  accessible={false}
  collapsable={false}
  focusable={false}
  onAccessibilityAction={[Function]}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "borderColor": undefined,
      "borderRadius": undefined,
      "borderWidth": undefined,
      "flexDirection": "row-reverse",
      "minHeight": undefined,
      "minWidth": undefined,
    }
  }
  switchAnimationStyles={
    Object {
      "thumbAnimatedStyle": Object {
        "transform": Array [
          Object {
            "translateX": 0,
          },
        ],
      },
      "trackBackgroundStyle": Object {
        "backgroundColor": "rgba(240, 240, 240, 1)",
      },
    }
  }
>
  <Text
    style={
      Object {
        "color": "#e0e0e0",
        "fontFamily": "Segoe UI",
        "fontSize": 13,
        "fontWeight": "400",
        "margin": 0,
      }
    }
  >
    Default Switch Disabled
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row-reverse",
      }
    }
  >
    <View
      collapsable={false}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(240, 240, 240, 1)",
          "borderRadius": 100,
          "flexDirection": "row",
          "height": 32,
          "marginRight": 8,
          "width": 52,
        }
      }
    >
      <View
        animationClass="Ribbon_SwitchThumb"
        collapsable={false}
        style={
          Object {
            "backgroundColor": "#ffffff",
            "borderRadius": 100,
            "elevation": 0,
            "height": 26,
            "margin": 3,
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
            "width": 26,
          }
        }
      />
    </View>
  </View>
</View>
`;
